test/index.spec.ts

```diff
@@ -2,183 +2,180 @@
+    ['npm tst', 'yarn test', 'pnpm test'],
+    ['npm test -- --version', 'yarn test --version', 'pnpm test -- --version'],
+    ['npm test -- -v', 'yarn test -v', 'pnpm test -- -v'],
+    // unchanged
+    ['npm start', 'yarn start', 'pnpm start'],
+    ['npm stop', 'yarn stop', "npm stop\n# couldn't auto-convert command"],
+    // unsupported
+    [
+      'npm whoami',
+      "npm whoami\n# couldn't auto-convert command",
+      "npm whoami\n# couldn't auto-convert command"
+    ],
+    // init
+    ['npm init', 'yarn init', 'pnpm init'],
+    ['npm init -y', 'yarn init -y', 'pnpm init -y'],
+    ['npm init --yes', 'yarn init --yes', 'pnpm init --yes'],
+    ['npm init --scope', 'yarn init', 'pnpm init'],
+    ['npm init --private', 'yarn init --private', 'pnpm init --private'],
+    ['npm init --unknown-arg', 'yarn init --unknown-arg', 'pnpm init --unknown-arg'],
+    ['npm init esm --yes', 'yarn create esm --yes', 'pnpm create esm --yes'],
+    [
+      'npm init @scope/my-package',
+      'yarn create @scope/my-package',
+      'pnpm create @scope/my-package'
+    ],
+    [
+      'npm init react-app ./my-react-app',
+      'yarn create react-app ./my-react-app',
+      'pnpm create react-app ./my-react-app'
+    ],
+    // list
+    ['npm list', 'yarn list', 'pnpm list'],
+    ['npm ls', 'yarn list', 'pnpm ls'],
+    ['npm list --production', 'yarn list --production', 'pnpm list --prod'],
+    ['npm list --development', 'yarn list --development', 'pnpm list --dev'],
+    ['npm list --global', 'yarn list --global', 'pnpm list --global'],
+    ['npm list --depth=0', 'yarn list --depth=0', 'pnpm list --depth 0'],
+    ['npm list package', 'yarn list --pattern "package"', 'pnpm list package'],
+    [
+      'npm list package package2',
+      'yarn list --pattern "package|package2"',
+      'pnpm list package package2'
+    ],
+    [
+      'npm list @scope/package @scope/package2',
+      'yarn list --pattern "@scope/package|@scope/package2"',
+      'pnpm list @scope/package @scope/package2'
+    ],
+    [
+      'npm list @scope/package @scope/package2 --depth=2',
+      'yarn list --pattern "@scope/package|@scope/package2" --depth=2',
+      'pnpm list @scope/package @scope/package2 --depth 2'
+    ],
+    // link
+    ['npm ln', 'yarn link', 'pnpm link'],
+    ['npm ln package', 'yarn link package', 'pnpm link package'],
+    ['npm link', 'yarn link', 'pnpm link'],
+    ['npm link package', 'yarn link package', 'pnpm link package'],
+    // unlink
+    ['npm unlink', 'yarn unlink', 'pnpm unlink'],
+    ['npm unlink package', 'yarn unlink package', 'pnpm unlink --filter package']
+  ]
+
+  describe('to Yarn', () => {
+    it.each(tests)('%s', (npmValue, yarnValue) => {
+      expect(convert(npmValue, 'yarn')).toEqual(yarnValue)
+    })
+  })
+
+  describe('to PNPM', () => {
+    it.each(tests)('%s', (npmValue, _yarnValue, pnpmValue) => {
+      expect(convert(npmValue, 'pnpm')).toEqual(pnpmValue)
+    })
```

Comment:
I converted all tests to be executed in loop, to make sure that if we add support for tests we validate it against yarn and pnpm